core/src/main/java/org/infinispan/protostream/impl/TagWriterImpl.java

```diff
@@ -66,9 +67,13 @@ public static TagWriterImpl newNestedInstance(ProtobufTagMarshaller.WriteContext
    }
 
    public static TagWriterImpl newInstance(ImmutableSerializationContext serCtx, OutputStream output) {
-      return new TagWriterImpl((SerializationContextImpl) serCtx, new OutputStreamEncoder(output, ProtobufUtil.DEFAULT_STREAM_BUFFER_SIZE));
+      return new TagWriterImpl((SerializationContextImpl) serCtx, new OutputStreamNoBufferEncoder(output));
    }
 
+   /**
+    * @deprecated since 4.6.3 Please use {@link #newInstance(ImmutableSerializationContext, OutputStream)} with a {@link java.io.BufferedOutputStream instead}
+    */
+   @Deprecated
    public static TagWriterImpl newInstance(ImmutableSerializationContext serCtx, OutputStream output, int bufferSize) {
       return new TagWriterImpl((SerializationContextImpl) serCtx, new OutputStreamEncoder(output, bufferSize));
```

Comment:
I'm guessing that the `bufferSize` can be ignored and we can use `OutputStreamNoBufferEncoder`.